src/rebuild/srv.c

```diff
@@ -1308,11 +1308,12 @@ rebuild_task_ult(void *arg)
 			DP_UUID(task->dst_pool_uuid));
 
 	ds_pool_put(pool);
-	ABT_mutex_lock(rgt->rgt_lock);
-	ABT_cond_signal(rgt->rgt_done_cond);
-	ABT_mutex_unlock(rgt->rgt_lock);
-	if (rgt)
+	if (rgt) {
```

Comment:
Just to be clear, this change I proactively merged from  https://github.com/daos-stack/daos/pull/4564/  before that PR was landed to master (it has since landed but it is not worth merging master and running through CI testing in this patch).